sentry_sdk/integrations/asyncpg.py

```diff
@@ -66,8 +66,12 @@ async def _inner(*args: Any, **kwargs: Any) -> T:
             return await f(*args, **kwargs)
 
         query = args[1]
-        with record_sql_queries(hub, None, query, None, None, executemany=False):
+        with record_sql_queries(
+            hub, None, query, None, None, executemany=False
+        ) as span:
             res = await f(*args, **kwargs)
+
+        add_query_source(hub, span)
```

Comment:
Why is this outside the "with" block. Is here `span` existing?

Reply:
It's outside so that we only `add_query_source` once the span has finished and it has an end timestamp -- that way we now how long it took and can avoid looking for the frame for short spans.
`span` still points to the same span after the `with`. Same as loop variables.

Reply:
ha! did not know this :-)